nav2-usability/scenario_code/lib/skipping-stacks/skipping_stacks_go_router_prototype.dart

```diff
@@ -0,0 +1,218 @@
+// Copyright 2022, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:go_router_prototype/go_router_prototype.dart';
+
+void main() {
+  runApp(BooksApp());
+}
+
+class Book {
+  final String title;
+  final Author author;
+
+  Book(this.title, this.author);
+}
+
+class Author {
+  String name;
+
+  Author(this.name);
+}
+
+class AppState extends ChangeNotifier {
+  final List<Book> books = [
+    Book('Stranger in a Strange Land', Author('Robert A. Heinlein')),
+    Book('Foundation', Author('Isaac Asimov')),
+    Book('Fahrenheit 451', Author('Ray Bradbury')),
+  ];
+
+  List<Author> get authors => [...books.map((book) => book.author)];
+}
+
+class BooksApp extends StatelessWidget {
+  final AppState _appState = AppState();
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp.router(
+      routeInformationParser: _router.parser,
+      routerDelegate: _router.delegate,
+    );
+  }
+
+  late final _router = GoRouter(
+    routes: [
+      ShellRoute(
+        path: '/',
+        // Home just redirects to the list of books
+        defaultRoute: 'books',
+        builder: (context, child) => AnimatedSwitcher(
+          duration: Duration(milliseconds: 400),
+          child: child,
+        ),
+        routes: [
+          StackedRoute(
+            path: 'books',
+            builder: (context) => BooksListScreen(books: _appState.books),
+            routes: [
+              StackedRoute(
+                path: r':bookId',
```

Comment:
Can this be simplified?

```suggestion
                path: ':bookId',
```

Reply:
You're right, I don't see raw strings aren't needed here - fixed in [0aeec73](https://github.com/flutter/uxr/pull/82/commits/0aeec73bb02d8e48983b3e0b06ad3901bb9c5935). GoRouter still supports regex syntax like `: r'wishlist/:id(\d+)'` I think, but the prototype doesn't.